src/decode-udp.c

```diff
@@ -103,12 +103,18 @@ int DecodeUDP(ThreadVars *tv, DecodeThreadVars *dtv, Packet *p,
     }
 
     /* Handle VXLAN if configured */
-    if (DecodeVXLANEnabledForPort(p->sp, p->dp) &&
-            unlikely(DecodeVXLAN(tv, dtv, p, p->payload, p->payload_len) == TM_ECODE_OK)) {
-        /* Here we have a VXLAN packet and don't need to handle app
-         * layer */
-        FlowSetupPacket(p);
-        return TM_ECODE_OK;
+    if (DecodeVXLANEnabledForPort(p->sp, p->dp)) {
+        Packet *tp =
+                PacketTunnelPktSetup(tv, dtv, p, p->payload, p->payload_len, DECODE_TUNNEL_VXLAN);
```

Comment:
I fear this will lead to many more false positive packets. The reason for the current setup is that since this is parsing UDP datagrams, it may fp on random packets. And we've seen a lot of this. So we try to parse strictly first, only then definitively conclude we are in fact dealing with vxlan.

Reply:
Thanks, will change this